cmake/modules/FindQatDrv.cmake

```diff
@@ -0,0 +1,113 @@
+             "i.e. \"export ICP_ROOT=/the/directory/of/QAT\". "
+             "Or download and install QAT driver ${expected_version} manually at the link "
+             "${expected_version_url}. "
+             "Remember to set the environment variable \"ICP_ROOT\" for the package root directory.")
+  else()
+    set(source_dir $ENV{ICP_ROOT})
+    set(versionfile "${source_dir}/versionfile")
+  endif()
+	
+  string(REGEX REPLACE "^v1.7.L.([0-9]+.[0-9]+.[0-9]+).[0-9]+$"
+	 "\\1" EXPECTED_version "${expected_version}")
+  
+  file(STRINGS "${versionfile}" QAT_VERSION_LINE
+       REGEX "^PACKAGE_TYPE=QAT[0-9]+.[0-9]+$")
+  string(REGEX REPLACE "^PACKAGE_TYPE=QAT([0-9]+.[0-9]+)$"
+         "\\1" QAT_VERSION1 "${QAT_VERSION_LINE}")
+  unset(QAT_VERSION_LINE)
+  file(STRINGS "${versionfile}" QAT_VERSION_LINE
+       REGEX "^PACKAGE_OS=[A-Z]$")
+  string(REGEX REPLACE "^PACKAGE_OS=([A-Z])$"
+         "\\1" QAT_VERSION2 "${QAT_VERSION_LINE}")
+  unset(QAT_VERSION_LINE)
+  file(STRINGS "${versionfile}" QAT_VERSION_LINE
+       REGEX "^PACKAGE_VERSION_MAJOR_NUMBER=[0-9]+$")
+  string(REGEX REPLACE "^PACKAGE_VERSION_MAJOR_NUMBER=([0-9]+)$"
+         "\\1" QAT_VERSION3 "${QAT_VERSION_LINE}")
+  unset(QAT_VERSION_LINE)
+  file(STRINGS "${versionfile}" QAT_VERSION_LINE
+       REGEX "^PACKAGE_VERSION_MINOR_NUMBER=[0-9]+$")
+  string(REGEX REPLACE "^PACKAGE_VERSION_MINOR_NUMBER=([0-9]+)$"
+         "\\1" QAT_VERSION4 "${QAT_VERSION_LINE}")
+  unset(QAT_VERSION_LINE)
+  file(STRINGS "${versionfile}" QAT_VERSION_LINE
+       REGEX "^PACKAGE_VERSION_PATCH_NUMBER=[0-9]+$")
+  string(REGEX REPLACE "^PACKAGE_VERSION_PATCH_NUMBER=([0-9]+)$"
+         "\\1" QAT_VERSION5 "${QAT_VERSION_LINE}")
+  unset(QAT_VERSION_LINE)
+  file(STRINGS "${versionfile}" QAT_VERSION_LINE
+       REGEX "^PACKAGE_VERSION_BUILD_NUMBER=[0-9]+$")
+  string(REGEX REPLACE "^PACKAGE_VERSION_BUILD_NUMBER=([0-9]+)$"
+         "\\1" QAT_VERSION6 "${QAT_VERSION_LINE}")
+  set(version_a "v${QAT_VERSION1}.${QAT_VERSION2}.${QAT_VERSION3}.${QAT_VERSION4}.${QAT_VERSION5}-${QAT_VERSION6}")
+  set(version "${QAT_VERSION3}.${QAT_VERSION4}.${QAT_VERSION5}")
+  if(version VERSION_LESS EXPECTED_version)
+    message(FATAL_ERROR "QAT ${version_a} in ${source_dir} is not new enough. "
+	    "Please either \"rm -rf ${source_dir}\" "
+            "so you can download and install QAT driver ${expected_version} manually at the link "
+	    "${expected_version_url}, or make sure ${source_dir} "
+            "contains a full copy of QAT driver ${expected_version}. "
+            "Remember to set an environment variable for the package root directory, "
+            "i.e. \"export ICP_ROOT=/opt/APP/driver/QAT\"." )
+  else()
+    message(STATUS "QAT driver (${version} >= ${EXPECTED_version}) already in ${source_dir}")
+  endif()
+endfunction()
+	
+check_qatdrv_version(${expected_version} ${expected_version_url})  
+
+set(QatDrv_INCLUDE_DIRS
```

Comment:
this is not quite right, IMHO. in general, this script is supposed to find the header file(s) and it sets `QatDrv_INCLUDE_DIRS` only if all of those header files are found.